### PR TITLE
default app logo changed and login screen refactored and redesign. Splash screen and login error screen added

### PR DESCRIPTION
**Pull request by Moises-dev08**
### Description:

- Set up RN for Android
- Set up for testing 
- Login screen redesign
- Splash screen created
- Login errors screen created
- Tests for login screen and error screen added

#### Tasks:

- [no] Tested on iOS
- [yes] Tested on Android
- [no] Tested on a small device

### CODE REVIEW

File: android/app/build.gradle [hunk 1]
```diff
@@ -252,3 +252,4 @@ task copyDownloadableDepsToLibs(type: Copy) {
 }
 
 apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
+apply plugin: 'com.google.gms.google-services'
```
Comment (brunopintos) [inline android/app/build.gradle:255]: why did we change the order of the apply plugin?

File: src/screens/LoginErrorScreen/LoginErrorScreen.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import Text from '../../components/Text';
+import {
+  SOMETHING_WENT_WRONG,
+  PLEASE_TRY_AGAIN,
+  CHECK_YOUR_INTERNET_CONNECTION,
+  TRY_AGAIN,
+} from '../../constants/screens';
+import errorAvatar from '../../assets/images/loginIcons/errorAvatar.png';
+import ohhMessage from '../../assets/images/loginIcons/Ohh...SH_T.png';
+import connectionErrorAvatar from '../../assets/images/loginIcons/connectionErrorAvatar.png';
```
Comment (brunopintos) [inline src/screens/LoginErrorScreen/LoginErrorScreen.js]: I think this project has absolute path, so you shouldn't do `../../`

When you import, just start from src folder, so the connectionErrorAvatar import, for example, would something like this:
```
import connectionErrorAvatar from 'assets/images/loginIcons/connectionErrorAvatar.png';
```

File: src/screens/LoginErrorScreen/__test__/loginErrorScreen.test.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+
+import LoginErrorScreen from '../LoginErrorScreen';
```
Comment (brunopintos) [inline src/screens/LoginErrorScreen/__test__/loginErrorScreen.test.js]: use absolute import

File: src/screens/LoginScreen/LoginScreen.styles.js [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
+    height: 78.12,
+  },
+  welcomeText: {
+    fontStyle: 'normal',
+  },
+  welcomeSubText: {
+    fontStyle: 'normal',
+    marginBottom: 12,
+  },
+  loginButton: {},
```
Comment (brunopintos) [inline src/screens/LoginScreen/LoginScreen.styles.js]: we could delete this empty object

File: src/screens/LoginScreen/__test__/loginScreen.test.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { render, fireEvent, act } from '@testing-library/react-native';
+import { Provider } from 'react-redux';
+import Config from 'react-native-config';
+import { createStore } from 'redux';
+
+import LoginScreen from '../LoginScreen';
+import { GoogleSignin } from '../../../../tests/__mocks__/@react-native-google-signin/google-signin';
+import isConnected from '../../../../tests/__mocks__/@react-native-community/netinfo';
+import reducer from '../../../reducers';
```
Comment (brunopintos) [inline src/screens/LoginScreen/__test__/loginScreen.test.js]: use absolute path here

File: src/screens/LoginScreen/__test__/loginScreen.test.js [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+test('should click in GoogleSignIn button', () => {
+  const { getByText } = render(
+    <Provider store={store}>
+      <LoginScreen />
+    </Provider>,
+  );
+
+  act(() => jest.advanceTimersByTime(1500));
+
+  const button = getByText(/Tu cuenta de Google/i);
```
Comment (brunopintos) [inline src/screens/LoginScreen/__test__/loginScreen.test.js]: use english language

File: src/screens/SplashScreen/SplashScreen.js [hunk 7]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { View, Image } from 'react-native';
+
+import appLogo from '../../assets/images/appLogo/default.png';
```
Comment (brunopintos) [inline src/screens/SplashScreen/SplashScreen.js]: use absolute imports here

File: src/utils/configureNotifications.js [hunk 8]
```diff
@@ -70,7 +70,7 @@ const configureNotifications = store => {
       store.dispatch(updateFirebaseToken(fcmToken));
     },
     onRegistrationError: err => {
-      console.log(err);
+      return err;
```
Comment (brunopintos) [inline src/utils/configureNotifications.js]: you could use implicit return here: `onRegistrationError: err => err`

File: tests/__mocks__/@react-native-community/netinfo.js [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export default {
+  getCurrentConnectivity: jest.fn(),
+  isConnectionMetered: jest.fn(),
+  addListener: jest.fn(),
+  removeListeners: jest.fn(),
+  isConnected: {
+    fetch: () => {
+      return Promise.resolve(true);
```
Comment (brunopintos) [inline tests/__mocks__/@react-native-community/netinfo.js]: you could use implicit return here: `fetch: () => Promise.resolve(true),`

**Review by brunopintos — CHANGES_REQUESTED** [review]
Some small changes to do.

Whe use absolute path for all of the imports, EXCEPT when we want to import a file that is in the same folder. In that case we use `import smth from './smth';`

File: android/app/build.gradle (hunk 1 above)
Comment (Moises-dev08) [reply android/app/build.gradle:255]: I changed it because it was the library advice. (react-native-google-signin/google-signin)

**Review by brunopintos — APPROVED** [review]
GTG

File: src/screens/LoginScreen/__test__/loginScreen.test.js (hunk 5 above)
Comment (brunopintos) [inline src/screens/LoginScreen/__test__/loginScreen.test.js]: Use absolute import like so:
```
import LoginScreen from 'screens/LoginScreen/LoginScreen';
import { GoogleSignin } from 'tests/__mocks__/@react-native-google-signin/google-signin';
import isConnected from 'tests/__mocks__/@react-native-community/netinfo';
import reducer from 'reducers';
```
I don't know if this works with all of the imports, but it should

Comment (Moises-dev08) [reply src/screens/LoginScreen/__test__/loginScreen.test.js]: yes it does

**Pull request merged**